File: easy.server.demo/easy.server.boot.demo/src/main/java/io/github/jinlongliao/easy/server/boot/demo/config/web/WebConfig.java
package io.github.jinlongliao.easy.server.boot.demo.config.web;

import io.github.jinlongliao.easy.server.boot.demo.config.swagger.DemoProxyAccessServlet;
import io.github.jinlongliao.easy.server.cached.aop.simple.SimplePointcutAndHandler;
import io.github.jinlongliao.easy.server.swagger.config.ApiConfig;
import io.github.jinlongliao.easy.server.utils.json.JsonHelper;
import org.springframework.beans.factory.ListableBeanFactory;
import org.springframework.context.annotation.Bean;

/**
 * @author: liaojinlong
 * @date: 2022-10-12 11:12
 */
public class WebConfig {
    @Bean
    public SpringServletBeanConfig servletBeanConfig() {
        return new SpringServletBeanConfig();
    }

    @Bean
    public DemoProxyAccessServlet demoProxyAccessServlet(ApiConfig apiConfig, JsonHelper jsonHelper) {
        return new DemoProxyAccessServlet(apiConfig, jsonHelper);
    }


}

File: easy.server.demo/easy.server.kotlin.demo/src/main/kotlin/io/github/jinlongliao/easy/server/kotlin/demo/config/web/WebConfig.kt
package io.github.jinlongliao.easy.server.kotlin.demo.config.web

import io.github.jinlongliao.easy.server.kotlin.demo.config.swagger.DemoProxyAccessServlet
import io.github.jinlongliao.easy.server.swagger.config.ApiConfig
import io.github.jinlongliao.easy.server.utils.json.JsonHelper
import org.springframework.context.annotation.Bean

/**
 * @author: liaojinlong
 * @date: 2022-10-12 11:12
 */
class WebConfig {
    @Bean
    fun servletBeanConfig(): SpringServletBeanConfig {
        return SpringServletBeanConfig()
    }

    @Bean
    fun demoProxyAccessServlet(apiConfig: ApiConfig?, jsonHelper: JsonHelper): DemoProxyAccessServlet {
        return DemoProxyAccessServlet(apiConfig, jsonHelper)
    }
}
